core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1039,7 +1039,9 @@ else if (prepared.containsKey(profile.getProfile())) {
     }
 
     protected Map<String, PrepareContractionHierarchies.Result> prepareCH(boolean closeEarly, List<CHConfig> configsToPrepare) {
-        ensureWriteAccess();
+        if (!closeEarly && !configsToPrepare.isEmpty()) { // do not check write access if no ch profiles have to be prepared and no ealry closing is set
+            ensureWriteAccess();
+        }
```

Comment:
I think we should deal with this situation at an earlier point. When `prepareCH` is called it means that CHs should be prepared (written). So if you meant to do a 'pure load' this should be prevented on a higher level. Either you need to take care of this in your configuration (do not add more CH profiles than there are on disk so no delta imports will run), or `loadOrPrepareCH` should offer something like `deltaImport=true`.

Reply:
Well that is exactely the problem: Although there are no additional CH profiles added, `loadOrPrepareCH` still calls `prepareCH(closeEarly, configsToPrepare)` with an empty `List<CHConfig> configsToPrepare`.

So we could check if `List<CHConfig> configsToPrepare` is empty before calling `prepareCH` altogether... however, inside of `prepareCH` there is some `closeEarly` functionality that flushes and closes the `ghStorage`, which is used when called via `importAndClose() -> process(closeEarly = true) -> postProcessing(closeEarly = true) -> loadOrPrepareCH(closeEarly = true) -> prepareCH(closeEarly = true, configsToPrepare)`.

If I prevent `prepareCH` from being called when there are no ` configsToPrepare`, then the `closeEarly` code is not applied.

Can the `closeEarly` code be ignored? Does it belong in `prepareCH`?

Reply:
Ok I see.

> Can the closeEarly code be ignored?  Does it belong in prepareCH?

We should probably not ignore it, but it definitely does not belong in `prepareCH`.